pkg/csi/service/vanilla/controller.go

```diff
@@ -1068,6 +1095,16 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
 					"failed to delete volumeInfo CR for volume: %q. Error: %+v", req.VolumeId, err)
 			}
 		}
+		// If this is multi-VC configuration, delete CnsVolumeInfo CR
+		if multivCenterCSITopologyEnabled && len(c.managers.VcenterConfigs) > 1 {
+			if volumeInfoService != nil {
```

Comment:
Do you really need `if volumeInfoService != nil ` check here? What if it is nil, we cannot silently succeed DeleteVolume without deleting the CR right? 

@divyenpatel Will FullSync take care of such instances where deletion of CR failed?

Reply:
we will not create `CnsVolumeInfo `CR for the case of `volumeInfoService = nil`, so we need to delete the CR only when `volumeInfoService `is nil

Reply:
`volumeInfoService = nil` will only occur when `multivCenterCSITopologyEnabled = true` and `len(c.managers.VcenterConfigs) == 1`.

Reply:
yes, will remove the check `if volumeInfoService != nil` as it is not needed here